common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -7460,6 +7460,12 @@
     ],
     "sqlState" : "22023"
   },
+  "VECTOR_DIMENSION_MISMATCH" : {
+    "message" : [
+      "For the <functionName> function, both vectors must have the same dimension, but got <leftDim> and <rightDim>."
```

Comment:
nit: `Vectors passed to <functionName> must have the same dimension, but got <leftDim> and <rightDim>.`